carbon-design-system/carbon
# fix(feature-flags): update enabled to check for process

Closes #8007

This PR updates our `feature-flags` package to work correctly in environments that don't have `process.env` available. It adds in a try/catch block that will read from `process.env` and fallback to default values if it is undefined.

#### Changelog

**New**

**Changed**

- Update build task for `feature-flags` package to safely read from `process.env`

**Removed**

#### Testing / Reviewing

- Run build on the feature flags package 
- Verify output uses a try/catch around `process` statements/expressions
- Verify that the `enabled` setting for a flag correctly reads off `process.env` if it is defined, otherwise it should fall back to the default value
- Verify that the catch block uses the default enabled value for the flag
- Verify that the feature flags read the `enabled` value with the correct key
